src/client.cpp

```diff
@@ -881,15 +881,15 @@ int Client<IoType, SwitchDataIntegrity, SwitchActivityInfo, SwitchCycleDuration,
             if (p_client_bind_addr->ss_family != AF_UNSPEC) {
                 socklen_t client_bind_addr_len = g_pApp->m_const_params.client_bind_info_len;
                 std::string hostport = sockaddr_to_hostport(p_client_bind_addr);
-                struct sockaddr_store_t unix_addr;
-                socklen_t unix_addr_len;
                 if (p_client_bind_addr->ss_family == AF_UNIX && g_pApp->m_const_params.sock_type == SOCK_DGRAM) { // Need to bind localy
 #ifdef __windows__
                     log_err("AF_UNIX with DGRAM isn't supported in windows");
                     rc = SOCKPERF_ERR_SOCKET;
                     break;
 #else // __windows__
                     if (p_client_bind_addr->addr_un.sun_path[0] == 0) { // no specific addr client_info was provoided
+                        struct sockaddr_store_t unix_addr;
```

Comment:
Wow, we have non-trivial bug found by cppcheck:
```
error: invalidLifetime: /scrap/jenkins/workspace/sockperf/src/client.cpp:913: Using pointer to local variable 'unix_addr' that is out of scope.
```

Reply:
> Wow, we have non-trivial bug found by cppcheck:
> 
> ```
> error: invalidLifetime: /scrap/jenkins/workspace/sockperf/src/client.cpp:913: Using pointer to local variable 'unix_addr' that is out of scope.
> ```

`p_client_bind_addr = &unix_addr;` ?
Yep, good catch